transport/http/outbound.go

```diff
@@ -48,8 +48,14 @@ func NewOutbound(url string) transport.Outbound {
 }
 
 // NewOutboundWithClient builds a new HTTP outbound that sends requests to the
-// given URL using the given HTTP client.
+// given URL using the given HTTP client. If the client is nil, a new one is
+// assigned.
 func NewOutboundWithClient(url string, client *http.Client) transport.Outbound {
+	if client == nil {
+		// Instead of using a global client for all outbounds, we use an HTTP
+		// client per outbound if unspecified.
+		client = &http.Client{Transport: &http.Transport{}}
```

Comment:
Agreed that making them configurable is a must. At some point before we release 1.0, we should discuss what the defaults should be.

Even though they're wildly inappropriate, comparatively few people are setting these options properly for their raw http clients. As the authors of a fairly opinionated RPC framework, we should make the defaults appropriate for the common use case.